src/test/java/org/codehaus/plexus/archiver/zip/ArchiveFileComparator.java

```diff
@@ -1,202 +1,169 @@
-            {
-                final String name1 = ze1.getName();
-                final String name2 = prefix == null ? name1 : ( prefix + name1 );
-                TarArchiveEntry ze2 = map2.remove( name2 );
-                Assert.assertNotNull( ze2 );
-                assertEquals( file1, ze1, file2, ze2 );
-
-            }
 
+        forEachTarArchiveEntry( file1, entry1 -> {
+            final String name2 = prefix == null ? entry1.getName() : prefix + entry1.getName();
+            TarArchiveEntry entry2 = map2.remove( name2 );
+            assertThat( entry2 ).isNotNull();
+            assertTarEquals( file1, entry1, file2, entry2 );
         } );
-        Assert.assertTrue( map2.isEmpty() );
+        assertThat( map2 ).isEmpty();
     }
 
-    public static void assertEquals( final ArchiveFile file1, final ZipFile file2, final String prefix )
-        throws Exception
+    /**
+     * Called to compare the given Tar and Zip files.
+     */
+    public static void assertTarZipEquals( final TarFile file1, final ZipFile file2, final String prefix )
     {
-        final Map<String, ZipArchiveEntry> map2 = getFileEntries( file2 );
-        forEachTarArchiveEntry( file1, new TarArchiveEntryConsumer()
-        {
-
-            public void accept( TarArchiveEntry ze1 )
-                throws IOException
-            {
-                final String name1 = ze1.getName();
-                final String name2 = prefix == null ? name1 : ( prefix + name1 );
-                ZipArchiveEntry ze2 = map2.remove( name2 );
-                Assert.assertNotNull( ze2 );
-                assertEquals( file1, ze1, file2, ze2 );
-            }
-
+        forEachTarArchiveEntry( file1, entry1 -> {
+            final String name2 = prefix == null ? entry1.getName() : prefix + entry1.getName();
+            ZipArchiveEntry entry2 = file2.getEntry( name2 );
+            assertThat( entry2 ).isNotNull();
+            assertTarZipEquals( file1, entry1, file2, entry2 );
         } );
-        Assert.assertTrue( map2.isEmpty() );
     }
 
-    public static void assertEquals( org.apache.commons.compress.archivers.zip.ZipFile file1,
-                                     org.apache.commons.compress.archivers.zip.ZipFile file2, String prefix )
-        throws Exception
+    /**
+     * Called to compare the given Zip files.
+     */
+    public static void assertZipEquals( final ZipFile file1, final ZipFile file2, final String prefix )
     {
-        final Map map1 = getFileEntries( file1 );
-        final Map map2 = getFileEntries( file2 );
-        for ( Object o : map1.keySet() )
-        {
-            final String name1 = (String) o;
-            final String name2 = prefix == null ? name1 : ( prefix + name1 );
-            ZipArchiveEntry ze1 = (ZipArchiveEntry) map1.get( name1 );
-            ZipArchiveEntry ze2 = (ZipArchiveEntry) map2.remove( name2 );
-            Assert.assertNotNull( "Missing " + name1 + "in file 2", ze2 );
```

Comment:
I would keep the message as it would make it quicker to understand why the test is failing.